### PR TITLE
Get the REPL working

### PR DESCRIPTION
**Pull request by GingerIndustries**
We don't need anything fancy, just get it able to run Vyxal code. Closes Vyxal/Vyxal#1445.

### CODE REVIEW

**Comment by Steffan153** [comment]
If my prompt is empty and I press enter, it doesn't add a newline
image

**Comment by Steffan153** [comment]
If I type one letter and then press backspace, it erases the `vyxal>` prompt

**Comment by GingerIndustries** [comment]
@Steffan153 yeah, that's just one of a collection of Exciting Bugs I'm working on fixing

File: crosshatch/__init__.py
Comment (Steffan153) [inline crosshatch/__init__.py]: help

```suggestion
```

Comment (GingerIndustries) [reply crosshatch/__init__.py]: Already reverted in 341349da598725fd0626b0e8040c4d932c76a526 :p

Comment (Steffan153) [reply crosshatch/__init__.py]: i wasn't serious lol you can use the other rickroll

Comment (GingerIndustries) [reply crosshatch/__init__.py]: @Steffan153 It is, to say the least, not a good idea to curl random scripts from the internet

**Comment by GingerIndustries** [comment]
So, we ended up doing it fancy anyway :p Ah well, I'm not complaining!

File: pyproject.toml
Comment (Lyxal) [inline pyproject.toml:9]: 2.19

**Review by Lyxal — APPROVED** [review]
Lgtm

File: pyproject.toml
Comment (GingerIndustries) [reply pyproject.toml:9]: Huh, I thought I fixed that :p

File: pyproject.toml
Comment (ysthakur) [inline pyproject.toml:13]: Use black, it's great. Among other things, black will add a blank line to the end of all your files.

File: crosshatch/commands.py
Comment (ysthakur) [inline crosshatch/commands.py:39]: I hate snake case too but since it's Python, it's better if everything used snake case.

File: crosshatch/highlighter.py
Comment (ysthakur) [inline crosshatch/highlighter.py:21]: Why not just make these part of the original set and repeat `"bright_magenta"`? Or get rid of `vyxal.list_sep` and `vyxal.list_end` and just `vyxal.list` for separators and list closers.

**Review by ysthakur — CHANGES_REQUESTED** [review]
lgtm, but could we have a screen capture? I'm too lazy to install this and test it out myself :P

File: pyproject.toml
Comment (GingerIndustries) [reply pyproject.toml:13]: > lgtm, but could we have a screen capture? I'm too lazy to install this and test it out myself :P

Sure, here you go!
Screenshot_20230124_053241



> Use black, it's great. Among other things, black will add a blank line to the end of all your files.

Can't be bothered :b
Maybe it's possible to set up a workflow to auto-Blacken the code? IDK.

Comment (ysthakur) [reply pyproject.toml:13]: Yeah, Vyxal uses that pre-commit ci thing to automatically do it

Comment (ysthakur) [reply pyproject.toml:13]: I love those colors btw. Any way to have different highlighting for named function definitions and that `code` (I assume that's parsed as 4 commands)?

Comment (GingerIndustries) [reply pyproject.toml:13]: > I love those colors btw. Any way to have different highlighting for named function definitions and that `code` (I assume that's parsed as 4 commands)?

Probably, but you've just made me notice another bug: That `code` shouldn't be highlighted at all! Just the lambda and semicolon.

**Review by ysthakur — DISMISSED** [review]
I don't understand, why have separate `dev` and `main` branches if `main` is going to be the same as `dev` anyway? I thought the separation was so that a human could check that `dev` wasn't broken and then manually trigger merging into `main`.

File: crosshatch/commands.py
Comment (GingerIndustries) [reply crosshatch/commands.py:39]: I don't really want to do that though.

File: crosshatch/highlighter.py
Comment (GingerIndustries) [reply crosshatch/highlighter.py:21]: > Why not just make these part of the original set and repeat `"bright_magenta"`? Or get rid of `vyxal.list_sep` and `vyxal.list_end` and just use `vyxal.list` for separators and list closers too.

I'd like to not have duplicates, but unfortunately regex doesn't allow duplicate group names (which are what `rich` uses  for syntax highlighting).

**Comment by ysthakur** [comment]
@GingerIndustries I reviewed it when automerge was still there, which is why I thought you were going to have changes to dev automatically get merged into main. Dismissed the review but I don't know enough about Python or REPLs or Vyxal so I'm going to hold off on approving

**Comment by GingerIndustries** [comment]
@ysthakur cool! I'm going to merge tomorrow if nobody has any more objections.

**Review by Steffan153 — APPROVED** [review]
This is awesome - tab completions would be nice though, like the online interpeter

**Comment by Steffan153** [comment]
wait, using option+delete or ctrl+backspace doens't delete a whole word, which is annoying. i can't use option+left/right either which is horrible

**Comment by Steffan153** [comment]
ctrl+c doesn't stop the process which is also terrible (it could also just discard the current line and move to a new one)

**Review by Lyxal — DISMISSED** [review]
image

`ki` should not be highlighted differently here.

**Comment by GingerIndustries** [comment]
> image
> 
> `ki` should not be highlighted differently here.

@mathcat4 here's a spicy regex issue for you to fix

**Review by ysthakur — APPROVED** [review]

**Review by mathcat4 — APPROVED** [review]
yes goodies

**Pull request merged**
